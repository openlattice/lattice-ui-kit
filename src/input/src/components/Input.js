import styled from 'styled-components';
import { NEUTRALS, PURPLES, RED_1 } from '../../../colors';

const Input = styled.input`
  display: flex;
  flex: 0 1 auto;
  box-sizing: border-box;
  background: ${NEUTRALS[8]} url(${props => (props.icon ? props.icon : null)}) no-repeat center right 10px;
  border-radius: 3px;
  border: solid 1px ${props => (props.invalid ? RED_1 : NEUTRALS[4])};
  box-shadow: 0;
  color: ${NEUTRALS[0]};
  font-size: 14px;
  height: 39px;
  line-height: 19px;
  padding: ${props => (props.icon ? '0px 32px 0px 10px' : '0px 10px')};
  text-overflow: ellipsis;
  width: 100%;
  transition: background-color 0.2s ease-in-out, border-color 0.2s ease-in-out;

  :hover {
    background-color: ${NEUTRALS[6]};
  }

  :focus {
    border: solid 1px ${PURPLES[1]};
    background-color: white;
    outline: none;
  }

  :disabled {
    background-color: ${NEUTRALS[8]};
    color: ${NEUTRALS[1]};
    cursor: not-allowed;
  }
`;

export default Input;
