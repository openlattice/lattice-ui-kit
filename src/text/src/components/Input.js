// @flow

import styled from 'styled-components';
import inputStyles from './styled/inputStyles';

const Input = styled.input`
  ${inputStyles}
`;

export default Input;
