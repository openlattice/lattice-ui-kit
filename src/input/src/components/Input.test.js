import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from './Input';

describe('Input', () => {

  test('render matches snapshot', () => {
    const wrapper = shallow(<Input />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('native input props', () => {
    test('should pass through any props to the input', () => {
      const nativeProps = {
        type: 'text',
        disabled: true,
        name: 'test',
        placeholder: 'test placeholder',
        maxLength: 8,
        min: 1,
        max: 8,
        required: true,
        autoComplete: 'on',
        form: 'test-form',
        pattern: '/.+/',
      };

      const wrapper = mount(<Input {...nativeProps} />)
        .find('input')
        .props();
      expect(wrapper).toEqual(expect.objectContaining(nativeProps));
    });

  });

  describe('render icon', () => {
  });

});
