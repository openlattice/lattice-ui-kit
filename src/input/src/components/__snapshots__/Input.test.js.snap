// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  box-sizing: border-box;
  border-radius: 3px;
  box-shadow: 0;
  color: #555e6f;
  font-size: 14px;
  line-height: 18px;
  text-overflow: ellipsis;
  width: 100%;
  border: 1px solid #dcdce7;
  padding: 10px 10px;
  -webkit-transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out;
}

.c0:hover {
  background-color: #f0f0f7;
}

.c0:focus {
  border: solid 1px #6124e2;
  background-color: white;
  outline: none;
}

.c0:disabled {
  background-color: #f9f9fd;
  color: #8e929b;
  cursor: not-allowed;
}

<Input>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Input-sc-4g0udv-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:flex;flex:0 1 auto;box-sizing:border-box;border-radius:3px;box-shadow:0;color:",
            "#555e6f",
            ";font-size:14px;line-height:18px;text-overflow:ellipsis;width:100%;border:1px solid ",
            [Function],
            ";padding:10px 10px;transition:background-color 0.2s ease-in-out,border-color 0.2s ease-in-out;:hover{background-color:",
            "#f0f0f7",
            ";}:focus{border:solid 1px ",
            "#6124e2",
            ";background-color:white;outline:none;}:disabled{background-color:",
            "#f9f9fd",
            ";color:",
            "#8e929b",
            ";cursor:not-allowed;}",
          ],
        },
        "displayName": "Input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Input-sc-4g0udv-0",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <input
      className="c0"
    />
  </StyledComponent>
</Input>
`;
